flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -654,6 +661,19 @@ private void startTriggeringCheckpoint(CheckpointTriggerRequest request) {
                                         }
                                         return null;
                                     }));
+
+            pendingCheckpointCompletableFuture.thenAccept(
+                    pendingCheckpoint -> {
+                        synchronized (lock) {
+                            if (!pendingCheckpoint.isDisposed()) {
+                                pendingCheckpointPromises.put(
+                                        pendingCheckpoint.getCheckpointID(), checkpointComplete);
+                            } else {
+                                // what TODO?
+                                checkpointComplete.completeExceptionally(new RuntimeException());
```

Comment:
It looks that this only happens when the pending checkpoint got aborted right after it is created. Should probably use `PendingCheckpoint.getFailureCause()` instead.